src/engine/strat_engine/filesystem.rs

```diff
@@ -109,6 +153,30 @@ impl Recordable<FilesystemSave> for StratFilesystem {
     }
 }
 
+/// Return total blocks, free blocks, blocksize for a filesystem
+pub fn usage(mount_point: &PathBuf) -> EngineResult<(u64, u64, u64)> {
+    let mut stat = vfs::Statvfs::default();
+    try!(statvfs(mount_point, &mut stat));
+    Ok((stat.f_blocks, stat.f_bfree, stat.f_bsize))
```

Comment:
Ok.  I made these updates and it is better. On Monday I'll have another look at check().  I think you are right that the "shenanigans" should be more clear... I think it could be simplified.